spec/client_spec.rb

```diff
@@ -293,6 +293,31 @@ def is_include(array, attr, value)
     index.delete_index!
   end
 
+  it "should copy parts of the index only" do
+    index = Algolia::Index.new(safe_index_name("àlgol?à"))
+    begin
+      @index.clear_index
+      Algolia.delete_index! index.name
+    rescue
+      # friends_2 does not exist
+    end
+
+    @index.add_object!({:firstname => "Robert"})
+    @index.batch_synonyms! [
+      { :objectID => 'city', :type => 'synonym', :synonyms => ['San Francisco', 'SF'] },
+      { :objectID => 'street', :type => 'altCorrection1', :word => 'street', :corrections => ['st'] }
+    ]
+    @index.search('')['nbHits'].should eq(1)
+    @index.search_synonyms('')['nbHits'].should eq(2)
+
+    res = Algolia.copy_index!(safe_index_name("àlgol?a"), safe_index_name("àlgol?à"), ["synonyms"])
+
+    @index.delete_index!
+
```

Comment:
Shouldn't you check here that `@index.search('')['nbHits'].should eq(0)` ?

Reply:
:D nice catch. Indeed that's what I did at first but this is triggering an error from the API due to a weird behavior that won't be fixed.